src/js/providers/utils/quality-labels.ts

```diff
@@ -32,29 +44,31 @@ export function createLabel(height, bandwidth, redundant) {
 
 // Ensures that we're able to find a custom label. As long as there is at least 1 quality label and a defined
 // bandwidth, a quality label will always be found. Return null otherwise
-export function getCustomLabel(qualityLabels, bandwidth) {
-    let label = null;
+export function getCustomLabel(qualityLabels: QualityLabels | null, bandwidth?: number): string | null {
+    let label: string | null = null;
 
     if (bandwidth && qualityLabels) {
         const bandwidths = Object.keys(qualityLabels);
-        const key = parseFloat(bandwidth);
-        if (bandwidths.length && !isNaN(key)) {
-            label = qualityLabels[findClosestBandwidth(bandwidths, toKbps(key))];
+        if (bandwidths.length) {
+            const closestBandwidth = findClosestBandwidth(bandwidths, toKbps(bandwidth));
+            if (closestBandwidth) {
+                label = qualityLabels[closestBandwidth];
+            }
         }
     }
 
     return label;
 }
 
 // Finds the bandwidth with the smallest difference from the target bandwidth
-export function findClosestBandwidth(bandwidths, targetBandwidth) {
-    let closest = null;
+export function findClosestBandwidth(bandwidths: string[], targetBandwidth: number): string | null {
+    let closest: string | null = null;
     let smallestDiff = Infinity;
-    let curDiff;
+    let curDiff: number;
 
     if (Array.isArray(bandwidths)) {
```

Comment:
Don't need the `Array.isArray` check anymore- This function is only used above with:
```
const bandwidths = Object.keys(qualityLabels);
     if (bandwidths.length) {
        const closestBandwidth = findClosestBandwidth(bandwidths, toKbps(bandwidth));
```